tensorflow/core/grappler/costs/graph_properties.cc

```diff
@@ -1357,18 +1362,20 @@ class SymbolicShapeRefiner {
   // Returns true if all the output tensors have known values.
   bool AllOutputValuesKnown(NodeContext* c) {
     InferenceContext* ic = c->inference_context.get();
-    if (c->output_tensors_as_shapes.size() < ic->num_outputs() &&
-        c->output_tensor_protos.size() < ic->num_outputs()) {
+    int c_output_tensors_as_shapes_size = c->output_tensors_as_shapes.size();
+    int c_output_tensor_protos_size = c->output_tensor_protos.size();
```

Comment:
I would move these two on the `else` block to only be defined if that branch is taken. Eventually even move them inside the `for` there.

Reply:
They are actually used in before the `else` block, in the `if`.